apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/components/BtcAmountInput.tsx

```diff
@@ -0,0 +1,145 @@
+  tradeBtcPriceAtom,
+} from '../atoms'
+import {checkInputValueIsDecimal} from '../../../utils'
+import {getTokens, Stack} from 'tamagui'
+import {type DropdownItemProps, Dropdown} from '../../../../Dropdown'
+import {type BtcOrSat} from '../../../domain'
+import AmountInput from './AmountInput'
+import CalculatedWithLiveRate from './CalculatedWithLiveRate'
+
+const styles = StyleSheet.create({
+  dropdown: {
+    width: 65,
+  },
+  dropdownContainerStyle: {
+    backgroundColor: getTokens().color.greyAccent1.val,
+    borderRadius: getTokens().radius[4].val,
+    width: 100,
+    borderWidth: 0,
+    paddingTop: 10,
+    paddingBottom: 10,
+  },
+  dropdownItemContainerStyle: {
+    borderRadius: getTokens().radius[4].val,
+  },
+  selectedTextStyle: {
+    color: getTokens().color.white.val,
+    fontWeight: '500',
+    fontSize: 18,
+    fontFamily: 'TTSatoshi500',
+  },
+})
+
+const btcOrSatOptions: BtcOrSat[] = ['BTC', 'SAT']
+const BTC_INPUT_PLACEHOLDER = '0.05'
+
+interface Props {
+  automaticCalculationDisabled?: boolean
+  editable?: boolean
+  fiatValueAtom: PrimitiveAtom<string | undefined>
+  btcValueAtom: PrimitiveAtom<string | undefined>
+}
+
+function BtcAmountInput({
+  automaticCalculationDisabled,
+  btcValueAtom,
+  editable = true,
+  fiatValueAtom,
+}: Props): JSX.Element {
+  const ref = useRef<TextInput>(null)
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+
+  const [btcValue, setBtcValue] = useAtom(btcValueAtom)
+  const setFiatValue = useSetAtom(fiatValueAtom)
+  const feeAmount = useAtomValue(feeAmountAtom)
+  const [btcOrSat, setBtcOrSat] = useAtom(btcOrSatAtom)
+  const tradeBtcPrice = useAtomValue(tradeBtcPriceAtom)
+
+  const btcOrSatDropdownData: Array<DropdownItemProps<BtcOrSat>> = useMemo(
+    () =>
+      btcOrSatOptions.map((option) => ({
+        label: option,
+        value: option,
+      })),
+    []
+  )
+
+  const calculateFiatValueOnBtcAmountChange = useCallback(
+    (value: string | undefined) => {
+      setBtcValue(value)
+      if (!automaticCalculationDisabled) {
+        if (tradeBtcPrice && value) {
+          const numberValue =
+            btcOrSat === 'BTC' ? Number(value) : Number(value) / SATOSHIS_IN_BTC
+          const fee =
+            Math.round(tradeBtcPrice * numberValue) * (feeAmount / 100)
+          setFiatValue(`${Math.round(tradeBtcPrice * numberValue + fee)}`)
+        } else {
+          setFiatValue(undefined)
+        }
+      }
+    },
+    [
+      automaticCalculationDisabled,
+      btcOrSat,
+      feeAmount,
+      setBtcValue,
+      setFiatValue,
+      tradeBtcPrice,
+    ]
+  )
+
+  return (
+    <AmountInput
+      ref={ref}
+      editable={editable}
+      isFocused={isFocused}
+      onBlur={() => {
+        setIsFocused(false)
+      }}
+      onFocus={() => {
+        setIsFocused(true)
+      }}
+      onWrapperPress={() => {
+        ref.current?.focus()
+      }}
+      placeholder={BTC_INPUT_PLACEHOLDER}
+      value={btcValue}
```

Comment:
This will result in undefined being passed to input element. It is not recommended to change uncontrolled component to controlled one